packages/template-retail-react-app/app/hooks/use-active-data.test.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/*global dw*/
+/* eslint-disable no-import-assign */
+/* eslint-disable react-hooks/rules-of-hooks */
+import useActiveData from '@salesforce/retail-react-app/app/hooks/use-active-data'
+import {
+    mockCategory,
+    mockProduct,
+    mockSearchResults
+} from '@salesforce/retail-react-app/app/hooks/einstein-mock-data'
+import * as constants from '@salesforce/retail-react-app/app/constants'
+import {DEFAULT_SEARCH_PARAMS} from '@salesforce/retail-react-app/app/constants'
+
+jest.mock('@salesforce/retail-react-app/app/constants', () => {
+    const original = jest.requireActual('@salesforce/retail-react-app/app/constants')
+    return {
+        __esModule: true,
+        ...original,
+        ACTIVE_DATA_ENABLE: true
+    }
+})
+
+const activeDataApi = useActiveData()
+window.dw = class dw {}
+dw.ac = {
+    applyContext: jest.fn(),
+
+    _capture: jest.fn(),
+
+    _scheduleDataSubmission: jest.fn(),
+
+    _setSiteCurrency: jest.fn(),
+
+    setDWAnalytics: jest.fn()
+}
+
+dw.__dwAnalytics = {
+    getTracker: jest.fn()
+}
```

Comment:
```suggestion
beforeAll(() => {
  window.dw = {
    ac: {
      applyContext: jest.fn(),
      _capture: jest.fn(),
      _scheduleDataSubmission: jest.fn(),
      _setSiteCurrency: jest.fn(),
      setDWAnalytics: jest.fn()
    },
    __dwAnalytics: {
      getTracker: jest.fn()
    }
})
afterAll(() => {
  delete window.dw
})
```
1. `window.dw = class dw {}` is weird, especially given the actual implementation uses a plain object
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/fddd9b2c2c480638249822dfc7924bbf59017afe/packages/template-retail-react-app/app/static/head-active_data.js#L1
2. To avoid accidentally impacting unrelated tests, the setup/teardown should be in `beforeAll`/`afterAll` blocks.